src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidationContext.cs

```diff
@@ -59,8 +53,13 @@ public class OpenIdConnectProtocolValidationContext
         public string State { get; set; }
 
         /// <summary>
-        /// Gets or sets a validated id_token.
+        /// Gets or sets the response received from userinfo_endpoint.
+        /// </summary>
+        public JwtSecurityToken UserInfoEndpointResponse { get; set; }
```

Comment:
If response is signed, it is a jwt. Otherwise its a json (key value pairs) which is translated to JwtPayload. http://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse